src/Native/configure.cmake

```diff
@@ -226,6 +226,39 @@ check_include_files(
     linux/rtnetlink.h
     HAVE_LINUX_RTNETLINK_H)
 
+# getdomainname on OSX takes an 'int' instead of a 'size_t'
+# check if compiling with 'size_t' would cause a warning
+set (PREVIOUS_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
+set (CMAKE_REQUIRED_FLAGS "-Weverything")
+check_cxx_source_compiles(
+    "
+    #include <unistd.h>
+	int main() { size_t namelen = 20; char name[namelen]; getdomainname(name, namelen); }
+    "
+    HAVE_GETDOMAINNAME_SIZET
+	FAIL_REGEX "-Wshorten"
+)
+set (CMAKE_REQUIRED_FLAGS ${PREVIOUS_CMAKE_REQUIRED_FLAGS})
```

Comment:
That feels a little like cheating because check_prototype_definition fails on Linux because it can't find the definition, not because it doesn't match.